### ISSUE TITLE
Polygon2D.polygon is and is not a reference, but would be nice to be consistently a reference

### ISSUE THREAD

**Issue by lowagner** [godotengine/godot#91646]
### Tested versions

4.2.stable

### System information

Godot v4.2.stable - Pop!_OS 22.04 LTS - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 Laptop GPU () - 13th Gen Intel(R) Core(TM) i9-13900H (20 Threads)

### Issue description

per the documentation, Polygon2D.polygon is a copy, not a reference: https://docs.godotengine.org/en/stable/classes/class_polygon2d.html#class-polygon2d-property-polygon

i've noticed in code, however, that polygon vertices can be updated, but they can't be added/removed.  i didn't read the documentation closely when i first used Polygon2D.polygon, and have been happily updating `Polygon2D.polygon[0] = Vector2D(...first point...)`, etc., only to bump into weird issues when i try to change the number of points in the polygon (e.g., `Polygon2D.polygon.clear()` or `.resize(5)` or `.append(Vector2(...new point...))`).  i.e., that's when we start seeing the issue that `polygon` is *sometimes* a copy.

TL;DR: `Polygon2D.polygon.clear()` acts like clearing the copy (doesn't do what you want) and `Polygon2D.polygon[0]` acts like a reference to the first point, despite the documentation saying that `polygon` is not a reference.  IMO in an ideal world Polygon2D.polygon would be a reference so we could do both of these things and they do what we expect.

### Steps to reproduce

see the `wobble.gd` scene for using `polygon` as a reference (and it working) and the `ngon.gd` scene for trying to clear the points (and it not working if you use `bad_reset()`): https://github.com/lowagner/bad-ngon

### Minimal reproduction project (MRP)

https://github.com/lowagner/bad-ngon
